### PR TITLE
doc(device, e2e, prov-dev, service, samples): Fix javadoc issues

### PR DESCRIPTION
**Pull request by timtay-microsoft**
There are a number of javadoc issues that prevent successful GH-pages autogeneration. This PR fixes all outstanding issues

### CODE REVIEW

File: device/iot-device-samples/module-method-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ModuleMethodSample.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ else if (status == IotHubConnectionStatus.CONNECTED)
      * Receives method calls from IotHub. Default protocol is to use
      * use MQTT transport.
      *
-     * @param args 
+     * @param args The connection string and the selected protocol
```
Comment (bikamani) [inline device/iot-device-samples/module-method-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ModuleMethodSample.java]: Did you want to mention about the exceptions here instead?

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/provisioning/ProvisioningTests.java [hunk 2]
```diff
@@ -176,7 +176,8 @@ public void individualEnrollmentWithCustomAllocationPolicy() throws Exception
      * The custom allocation policy has a webhook to an Azure function, and that function will always dictate to provision
      * the device to the hub with the longest host name. This test verifies that an enrollment with a custom allocation policy
      * pointing to that Azure function will always enroll to the hub with the longest name
-     * @param enrollmentType
+     * @param enrollmentType The type of the enrollment to test
+     * @throws Exception if an exception occurs during provisioning or while
```
Comment (bikamani) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/provisioning/ProvisioningTests.java]: Or while?

File: device/iot-device-samples/module-method-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ModuleMethodSample.java (hunk 1 above)
Comment (timtay-microsoft) [reply device/iot-device-samples/module-method-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/ModuleMethodSample.java]: sure

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/provisioning/ProvisioningTests.java (hunk 2 above)
Comment (timtay-microsoft) [reply iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/provisioning/ProvisioningTests.java]: I'll finish the sentence, good catch

**Pull request merged**